raphtory-graphql/src/model/plugins/mutation_entry_point.rs

```diff
@@ -0,0 +1,26 @@
+use crate::model::algorithms::RegisterFunction;
+use async_graphql::dynamic::Object;
+use dynamic_graphql::internal::{OutputTypeName, Register, Registry, ResolveOwned, TypeName};
+use itertools::Itertools;
+use std::{collections::HashMap, sync::MutexGuard};
+
+pub trait MutationEntryPoint<'a>:
+Register + TypeName + OutputTypeName + ResolveOwned<'a> + Sync
+{
+    fn lock_plugins() -> MutexGuard<'static, HashMap<String, RegisterFunction>>;
```

Comment:
I assume this is called automatically and isn't up to the user to use?

Reply:
yeah it's only for internal use, maybe we should do this trait pub(crate), don't think anyone should be using it outside